### PR TITLE
Add documentation for Revision-related Service classes

### PR DESCRIPTION
**Pull request by brendajerop**
Add documentation for:
* RevisionUndoer
* RevisionRestorer
* RevisionDeleter
* RevisinPatroller
* RevisionRollbacker
* RevisionSaver

### CODE REVIEW

**Comment by samwilson** [comment]
All in one is probably fine, for documentation, if you're working on them all at the same time.

**Comment by brendajerop** [comment]
@samwilson This is ready for review

File: docs/revision_deleter.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   $revisionDeleter = $services->newRevisionDeleter();
+
+
+Deleting a revision
+-------------------
+
+.. code-block:: php
+
+   $revision = new \Mediawiki\DataModel\Revision( ... );
+   $isRevisionDeleted = $revisionDeleter->delete( revision );
```
Comment (samwilson) [inline docs/revision_deleter.rst]: Missing `$`.

File: docs/revision_patroller.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   $revisionPatroller = $services->newRevisionPatroller();
+
+
+Patrolling a revision
+---------------------
+
+.. code-block:: php
+
+   $revision = new \Mediawiki\DataModel\Revision( ... );
+   $isRevisionPatrolled = $revisionPatroller->patrol( revision );
```
Comment (samwilson) [inline docs/revision_patroller.rst]: Missing `$`.

File: docs/revision_restorer.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+   $revisionRestorer = $services->newRevisionRestorer();
+
+
+Restoring a revision
+--------------------
+
+.. code-block:: php
+
+   $revision = new \Mediawiki\DataModel\Revision( ... );
+   $isRevisionRestored = $revisionRestorer->restore( revision );
```
Comment (samwilson) [inline docs/revision_restorer.rst]: Missing `$`.

File: docs/revision_rollbacker.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+Rollback a revision
+-------------------
+
+.. code-block:: php
+
+   $revision = new \Mediawiki\DataModel\Revision( ... );
+   $title = new \Mediawiki\DataModel\Title( ... );
+   $isRevisionRollbacked = $revisionRollbacker->rollback( revision, title );
```
Comment (samwilson) [inline docs/revision_rollbacker.rst]: Missing `$`s.

**Pull request merged**